test/e2e/container_probe.go

```diff
@@ -41,9 +41,22 @@ var _ = Describe("Probing container", func() {
 	It("with readiness probe should not be ready before initial delay and never restart [Conformance]", func() {
 		p, err := podClient.Create(makePodSpec(probe.withInitialDelay().build(), nil))
 		expectNoError(err)
-		startTime := time.Now()
 
-		Expect(wait.Poll(poll, 90*time.Second, func() (bool, error) {
+		Expect(wait.Poll(poll, 120*time.Second, func() (bool, error) {
+			p, err := podClient.Get(p.Name)
+			if err != nil {
+				return false, err
+			}
+
+			if p.Status.Phase != api.PodRunning {
+				Logf("pod is not yet running; pod has phase %q.", p.Status.Phase)
+				return false, nil
+			}
+			return true, nil
+		})).NotTo(HaveOccurred(), "pod never became running")
+
+		startTime := time.Now()
```

Comment:
I'm worried this will be a little fragile now - there could be some time between when the pod became running (when the initial delay actually starts), and when we saw it as running (`time.Now()`). It would be more accurate to get the startTime from the containerStatus, but maybe that's too close to reimplementing the logic in the test?